### DISCUSSION TITLE
Understanding modbus

### DISCUSSION THREAD

**Discussion by mdeakin99** [wills106/homeassistant-solax-modbus#127] [Q&A]
Hi

Hopefully getting a solax inverter soon and presuming it will use pocket wifi and solax cloud and therefore need this integration for local access to data.

what I can figure out is

a) which rs485 ModBus adapter to buy
B) I don’t fully understand if I need to hardwire some cables or not.  Anyone got a picture/video of the ModBus adapter?

mark

**Comment by wills106** [comment]
Have you looked in the Wiki there is information on compatible adaptors and connection methods.

**Comment by mdeakin99** [comment]
Yep.  Meant to say I have already looked at the wiki but couldn’t find out if anything needs wiring or plug and play

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by wills106** [reply]
Most common adaptor is the Waveshare.

You need to do the initial Waveshare setup using a Windows PC (Setting IP address)
You can then configure it using the Web UI.

You can use an ethernet cable with one end chopped off to connect between the Inverter and the Waveshare (PIN 4 & 5)
Plug the Waveshare into your Network and apply power to it. Configure the Integration to look at the Waveshare within Home Assistant, the Integration should then configure itself based on which type of Inverter you. This is performed by reading the start of your Serial number.

**Comment by mdeakin99** [comment]
Would I be right in thinking


  1.  I wont have a Ethernet port for my network nearby so need to find once that includes a wifi mobule
  2.  These adapters need to be powered so I need to see if I can somehow plug it into a socket?

Sorry if these are dumb questions.

Thanks

Mark

**Reply by infradom** [reply]
I suggest and assume you will use a waveshare adapter:
The 2-wire modbus cable can be fairly long. So the Waveshare modbus-to-ethernet adapter can be placed in a location where you have Ethernet and power. Of course, you need this 2 wire cable (pin4 and 5) between the COM modbus port of the inverter (not to be confused with the meter modbus port) and the Waveshare
